lib/name_gender_classifier.rb

```diff
@@ -0,0 +1,104 @@
+require 'iconv'
+
+# Gender detector for first names.
+module NameGenderClassifier
+  # Return the gender(s) (probabilistically) for the informed name(s). The result type will vary depending
+  # on the parameter type:
+  #
+  # [String, Symbol] the gender (String) is returned.
+  # [Array<String>] an array (Array<String>) with the genders is returned.
+  # [Array<Object>] an array (Array<Object>) with the same objects and the newly assigned genders is returned.
+  #
+  # @param arg [String, Symbol, Array<String>, Array<Symbol>, Array<Object>] argument holding first
+  #   name(s) information(s).
+  # @paran options [Hash] first_name_attribute: name of the method that returns the first name,
+  #                       gender_attribute: name of the method which will receive the gender assignment.
+  #
+  # @return [String, Array<String>, Array<Object>] the gender classification for the passed first names
+  def self.classify(arg, options = {})
+    case arg
+    when String, Symbol
+      most_probable_gender(arg)
+    when Array
+      if arg[0].is_a?(String) || arg[0].is_a?(Symbol)
```

Comment:
Add a comment above line 23 that says:

# Assumes that all elements within the array are of the same type as the first.